addons/hr_expense/models/hr_expense.py

```diff
@@ -1054,7 +1054,8 @@ def _compute_expense_number(self):
     @api.depends('account_move_id.date')
     def _compute_accounting_date(self):
         for sheet in self:
-            sheet.accounting_date = sheet.account_move_id.date
+            default_data = self.env['ir.default'].get_model_defaults(self._name)
```

Comment:
as far as I can tell, this would also remove the fact that if we change the accounting date on the invoice, it will not be updated on the expense, which may not be wanted

in the given use case, the default is already computed from the default_get, we just pass here because we are modifying `account_move_id.date`

so maybe an alternative could be to not override the value with false if there is no account_move_id linked to the current expense

```diff
-            sheet.accounting_date = sheet.account_move_id.date
+            if sheet.account_move_id or not sheet.accounting_date:
+                sheet.accounting_date = sheet.account_move_id.date
```